File: isegoria/src/main/java/com/eulersbridge/isegoria/personality/PersonalityQuestionsFragment.java
package com.eulersbridge.isegoria.personality;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.eulersbridge.isegoria.R;
import com.eulersbridge.isegoria.network.api.models.UserPersonality;

public class PersonalityQuestionsFragment extends Fragment {

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.personality_screen2_fragment, container, false);

        //noinspection ConstantConditions
        PersonalityViewModel viewModel = ViewModelProviders.of(getActivity()).get(PersonalityViewModel.class);

        PersonalitySliderBar personalitySliderBar1 = rootView.findViewById(R.id.personalitySliderBar1);
        PersonalitySliderBar personalitySliderBar2 = rootView.findViewById(R.id.personalitySliderBar2);
        PersonalitySliderBar personalitySliderBar3 = rootView.findViewById(R.id.personalitySliderBar3);
        PersonalitySliderBar personalitySliderBar4 = rootView.findViewById(R.id.personalitySliderBar4);
        PersonalitySliderBar personalitySliderBar5 = rootView.findViewById(R.id.personalitySliderBar5);
        PersonalitySliderBar personalitySliderBar6 = rootView.findViewById(R.id.personalitySliderBar6);
        PersonalitySliderBar personalitySliderBar7 = rootView.findViewById(R.id.personalitySliderBar7);
        PersonalitySliderBar personalitySliderBar8 = rootView.findViewById(R.id.personalitySliderBar8);
        PersonalitySliderBar personalitySliderBar9 = rootView.findViewById(R.id.personalitySliderBar9);
        PersonalitySliderBar personalitySliderBar10 = rootView.findViewById(R.id.personalitySliderBar10);

        Button doneButton = rootView.findViewById(R.id.donePersonalityQuestions);
        doneButton.setOnClickListener(v -> {
            float extroversion = (personalitySliderBar1.getScore() + (8-personalitySliderBar6.getScore()))/2;
            float agreeableness = (personalitySliderBar7.getScore() + (8-personalitySliderBar2.getScore()))/2;
            float conscientiousness = (personalitySliderBar3.getScore() + (8-personalitySliderBar8.getScore()))/2;
            float emotionalStability = (personalitySliderBar9.getScore() + (8-personalitySliderBar4.getScore()))/2;
            float opennessToExperiences = (personalitySliderBar5.getScore() + (10-personalitySliderBar6.getScore()))/2;

            UserPersonality personality = new UserPersonality(agreeableness, conscientiousness,
                    emotionalStability, extroversion, opennessToExperiences);

            viewModel.setUserCompletedQuestions(personality).observe(this, success -> {
                if (success != null && !success) {
                    doneButton.post(() -> doneButton.setEnabled(true));
                }
            });
        });

        return rootView;
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/personality/PersonalityQuestionsFragment.kt
package com.eulersbridge.isegoria.personality

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.eulersbridge.isegoria.R
import com.eulersbridge.isegoria.network.api.models.UserPersonality
import com.eulersbridge.isegoria.observe
import kotlinx.android.synthetic.main.personality_screen2_fragment.*



class PersonalityQuestionsFragment : Fragment() {

    private lateinit var viewModel: PersonalityViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View?
            = inflater.inflate(R.layout.personality_screen2_fragment, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {

        val sliderBars = arrayOf(
            personalitySliderBar1, personalitySliderBar2, personalitySliderBar3,
            personalitySliderBar4, personalitySliderBar5, personalitySliderBar6,
            personalitySliderBar7, personalitySliderBar8, personalitySliderBar9,
            personalitySliderBar10
        )

        doneButton.setOnClickListener {
            val extroversion =
                ((sliderBars[0].score + (8 - sliderBars[5].score)) / 2).toFloat()
            val agreeableness =
                ((sliderBars[6].score + (8 - sliderBars[1].score)) / 2).toFloat()
            val conscientiousness =
                ((sliderBars[2].score + (8 - sliderBars[7].score)) / 2).toFloat()
            val emotionalStability =
                ((sliderBars[8].score + (8 - sliderBars[3].score)) / 2).toFloat()
            val opennessToExperiences =
                ((sliderBars[4].score + (10 - sliderBars[5].score)) / 2).toFloat()

            val personality = UserPersonality(
                agreeableness, conscientiousness,
                emotionalStability, extroversion, opennessToExperiences
            )

            observe(viewModel.setUserCompletedQuestions(personality)) { success ->
                if (success == false)
                    doneButton.post { doneButton.isEnabled = true }
            }
        }
    }

    fun setViewModel(viewModel: PersonalityViewModel) {
        this.viewModel = viewModel
    }
}